src/dbnode/integration/setup.go

```diff
@@ -468,11 +468,11 @@ func NewTestSetup(
 	for _, fn := range storageOptFns {
 		if fn != nil {
 			storageOpts = fn(storageOpts)
-			if storageOpts != nil && storageOpts.AdminClient() == nil {
-				storageOpts = storageOpts.SetAdminClient(adminClient)
-			}
 		}
 	}
+	if storageOpts != nil && storageOpts.AdminClient() == nil {
+		storageOpts = storageOpts.SetAdminClient(adminClient)
```

Comment:
I see that the value will not get set to `adminClient` if it is already != nil (which might probably be an indication of some setup issue?). Maybe, if it is not possible to modify the code to avoid this problem entirely, we should at least print a warning (something like "tried setting adminClient, but it was already set")?